chart/templates/etcd-statefulset.yaml

```diff
@@ -0,0 +1,174 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd-{{ .Values.role }}-client
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+spec:
+  type: ClusterIP
+  sessionAffinity: None
+  selector:
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+  ports:
+  - name: client
+    protocol: TCP
+    port: 2379
+    targetPort: 2379
+---
+apiVersion: {{ include "statefulsetversion" . }}
```

Comment:
Api version for Statefulset, in fact all the probable resources required for this helm chart are stable now. And not expected to change with k8s version. So, we should directly use it here, rather going via `versions.tpl` which only make sense in context of large projects like gardener.

Reply:
Done